python/cudf/cudf/core/series.py

```diff
@@ -1000,7 +1000,13 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         if fname in binary_operations:
             not_reflect = self is inputs[0]
             other = inputs[not_reflect]
-            op = f"__{'' if not_reflect else 'r'}{binary_operations[fname]}__"
+            op = binary_operations[fname]
+            ops_without_reflection = ("gt", "ge", "lt", "le", "ne", "eq")
+            if op in ops_without_reflection and not not_reflect:
+                # This replacement will ignore eq and ne, which don't need it.
+                op = op.replace(*(("g", "l") if "g" in op else ("l", "g")))
+                not_reflect = True
```

Comment:
`not_reflect` here is always `True`.

Reply:
Correct, that is intentional. The purpose of this block is that when one of these operations looks like it should be reflected (we only enter this block when `not_reflect` is `False`), instead of selecting a reflected op we change the operation to the inverse operation because no reflected op exists.

Reply:
I wasn't reading close enough, had meant `not_refect==True` before this line. Never mind!